Scripts/Plants/GrassManager.cs

```diff
@@ -95,50 +95,53 @@ where Valid(blockPos)
         }
 
         terrain.SetBlocks(blocksToChange);
-        blocks.AddRange(validBlocks);
+        blocks.UnionWith(validBlocks);
         time = 0;
         return true;
     }
 
     public override void LifeCycle(float delta)
     {
+        time += delta;
+        if (time < LIFECYCLE_TICK_TIME || blocks.Count == 0)
+            return;
+        time = 0;
+
         // remove blocks that are no longer grass
         List<IntVector3> blocksToRemove = (from block in blocks
                                            where terrain.GetBlock(block) != grassBlock
                                            select block).ToList<IntVector3>();
 
-        blocks = (from block in blocks
-                  where terrain.GetBlock(block) == grassBlock
-                  select block).ToList<IntVector3>();
+        blocks = new HashSet<IntVector3>(from block in blocks
+                                         where terrain.GetBlock(block) == grassBlock
+                                         select block);
 
-        time += delta;
-        if (time > LIFECYCLE_TICK_TIME && blocks.Count != 0)
-        {
-            time = 0;
+        // kill off some grass if there is too little gas
+        double numberToDie = 5*Math.Max(0.01 - atmosphere.GetGasAmt(Gas.NITROGEN), 0) +
+                             5*Math.Max(0.01 - atmosphere.GetGasAmt(Gas.CARBON_DIOXIDE), 0) +
+                             5*Math.Max(0.001 - atmosphere.GetGasAmt(Gas.OXYGEN), 0);
 
-            // kill off some grass if there is too little gas
-            float numberToDie = GAS_REQUIREMENTS.Sum(kvPair => 5 * Mathf.Max(kvPair.Value - atmosphere.GetGasAmt(kvPair.Key), 0));
```

Comment:
why don't you like the one-liner?

Reply:
Oh, I missed this, sorry (because I changed an older version). But I'm also not sure if it would be right to have it be exactly the same anyway. I don't mind changing it back for now though.

Reply:
what do you mean by "exactly the same anyway"?

Reply:
btw, even better would be to take the "5 * " in front of the sum

Reply:
Hmm I wonder if compiler would optimise this